singularity-eos/eos/eos.hpp

```diff
@@ -63,7 +64,7 @@ using singularity::detail::transform_variadic_list;
 
 // all eos's
 static constexpr const auto full_eos_list =
-    tl<IdealGas, Gruneisen, Vinet, JWL, DavisReactants, DavisProducts
+    tl<IdealGas, Gruneisen, Helmholtz, Vinet, JWL, DavisReactants, DavisProducts
```

Comment:
Should this be behind a preprocessor macro?

Reply:
Given the applicability is primarily targeted at astrophysical regimes, I think maybe this EOS should be opt in, meaning that all headers and implementation related files and tests should not be added to cmake headers, etc., or the type included in the variant, unless turned on. @Yurlungur thoughts?

Reply:
I think that makes sense @dholladay00 . I think my preference would be that the header file is always included, but tests are opt in and it's simply not included in the variant. Instead, downstream codes that expect to use it must add it to their own local variant.

Perhaps we should do the same for the stellar collapse tables.

Reply:
If this solution is not overly onerous for downstream codes that rely on these EOSs that is a reasonable solution, another good solution is something like what is done for eospac's inclusion in the variant.

Reply:
I think it's not too bad for them. Also given that astro codes are unlikely to want the terrestrial EOS'S and modifiers, it's maybe a good solution